src/api/transactions.js

```diff
@@ -444,26 +444,31 @@ export async function updateTransaction (ctx, transactionId) {
   const updates = ctx.request.body
 
   try {
+    const transaction = await TransactionModelAPI.findById(transactionId).exec()
+
     if (hasError(updates)) {
-      const transaction = await TransactionModelAPI.findById(transactionId).exec()
       const channel = await ChannelModelAPI.findById(transaction.channelID).exec()
       if (!autoRetryUtils.reachedMaxAttempts(transaction, channel)) {
         updates.autoRetry = true
         await autoRetryUtils.queueForRetry(transaction)
       }
     }
 
+    // construct promises array to remove all old payloads
+    const removeBodyPromises = promisesToRemoveAllTransactionBodies(transaction)
```

Comment:
I think we should remove bodies based on whats been updated. If only the response is updated, we'll end up with no request body

Reply:
Good point, ill update the function to analyse the update object and only remove the relevant bodies